docs/DesignDocs/libc_initialization.md

```diff
@@ -0,0 +1,169 @@
+# Musl libc Initialization
+
+## Normal musl libc initialization on Linux x86\_64
+
+A normal executable linux program with uses Musl libc will have the following initialization pattern,
+where `_start` is the entry point of every program.
+
+
+```
+_start()
+    v
+    |
+    --> _start_c()
+              v
+              |
+              --> _libc_start_main()
+                            v
+                            |
+                            --> __init_libc()
+                            --> _libc_start_main_stage2()
+                                          v
+                                          |
+                                          --> __libc_start_init()
```

Comment:
Can you expand this diagram to show when other GCC constructors and C++ constructors would be called.

Reply:
Sure. So that I remember for later, it is the last thing before calling `main`

Reply:
Actually, this is done in __libc_start_init